Добавление анимаций в проект

Nitraatss opened the pull request:
---
:mortar_board: [Добавление анимаций в проект](https://up.htmlacademy.ru/vue/1/user/334301/tasks/7)

Nitraatss committed:
6.4. Примеры анимаций с использованием CSS

Djaler commented:
Не очень красиво получается с анимацией при открытии окна входа - анимируется и подложка под ним тоже.
В идеале тут вот как сделать можно - отключать анимацию slide при переходе на страницу login, а анимировать только появление самого диалога. Ну и, наверное, ещё один отдельный маршрут для входа, чтоб на него переходить, когда мы находимся не в билдере

Djaler commented on src/frontend/src/layouts/AppLayoutDefault.vue:
```diff
@@ -2,7 +2,13 @@
 
-    <slot />
+    <transition
+      name="view"
```
чтобы не повторять в каждом лейауте - можно транзишен сделать в главном AppLayout

Djaler commented on src/frontend/src/layouts/AppLayoutInner.vue:
```diff
@@ -5,7 +5,13 @@
       <AppLayoutInnerSidebar />
 
-      <slot />
+      <transition
```
С этим лейаутом получается не очень красиво, что текст содержимого при анимации проходит над сайдбаром, хорошо бы наоборот

Nitraatss committed:
Исправление анимации появления ингредиентов

Nitraatss committed:
Правки

Исправлена анимация слайдов для страниц
Измено появление формы авторизации. При нажатии на вход не на странице конструктора идет переадресация на sign-in.
Добавлена информация об ошибках при валидации в форме авторизации.

Djaler commented on src/frontend/src/layouts/AppLayout.vue:
```diff
@@ -9,11 +12,37 @@ const defaultLayout = "AppLayoutDefault";
+  mounted() {
+    this.previousPath = this.$route.path;
+  },
+  updated() {
```
неплохое решение. Единственное, вместо updated правильнее было бы сделать вотчер на `$route`. Потому что updated вызывается при каждом ререндере компонента
